src/simulation/dynamics/spacecraftSystem/spacecraftSystem.cpp

```diff
@@ -1300,3 +1209,103 @@ void SpacecraftSystem::computeEnergyMomentumSystem(double time)
     
     return;
 }
+
+/*! Prepare for integration process, not currently implemented in SpacecraftSystem
+ @param integrateToThisTime Time to integrate to
+ */
+void SpacecraftSystem::preIntegration(double integrateToThisTime) {
+
+    // - Find the time step
+    this->timeStep = integrateToThisTime - this->timePrevious;
+
+    this->findPriorStateInformation(this->primaryCentralSpacecraft);
+
+    // - Call this for all of the unconnected spacecraft
+    std::vector<SpacecraftUnit*>::iterator spacecraftUnConnectedIt;
+    for(spacecraftUnConnectedIt = this->unDockedSpacecraft.begin(); spacecraftUnConnectedIt != this->unDockedSpacecraft.end(); spacecraftUnConnectedIt++)
+    {
+        this->findPriorStateInformation((*(*spacecraftUnConnectedIt)));
+    }
+
+    // - Integrate the state from the last time (timeBefore) to the integrateToThisTime
+    this->timeBefore = integrateToThisTime - this->timeStep;
+
+}
+
+/*! Perform post-integration steps, not currently implemented in SpacecraftSystem
+ @param integrateToThisTime Time to integrate to
+ */
+void SpacecraftSystem::postIntegration(double integrateToThisTime) {
+    std::vector<SpacecraftUnit*>::iterator spacecraftUnConnectedIt;
+
+    this->timePrevious = integrateToThisTime;     // - copy the current time into previous time for next integrate state call
+
+    // - Calculate the states of the attached spacecraft from the primary spacecraft
+    this->determineAttachedSCStates();
+
+    // - Call hubs modify states to allow for switching of MRPs
+    this->primaryCentralSpacecraft.hub.modifyStates(integrateToThisTime);
+
+    // - Just in case the MRPs of the attached hubs need to be switched
+    std::vector<SpacecraftUnit*>::iterator spacecraftConnectedIt;
+    for(spacecraftConnectedIt = this->spacecraftDockedToPrimary.begin(); spacecraftConnectedIt != this->spacecraftDockedToPrimary.end(); spacecraftConnectedIt++)
+    {
+        (*spacecraftConnectedIt)->hub.modifyStates(integrateToThisTime);
+    }
+
+    // - Just in case the MRPs of the attached hubs need to be switched
```

Comment:
I think this means to say "Just in case the MRPs of the UNattached hubs need to be switched"